### PR TITLE
Update licence

### PR DESCRIPTION
**Pull request by fe51**
Hi there, a PR to update the licence to apache 2.0, inspired by https://github.com/pyronear/pyro-vision/pull/132

It introduces the following changes : 
- updates licence to Apache 2.0
- updates headers
- adds script to check headers (copied from pyro-vision authored by frgfm)
- add ci job to check headers (copied from pyro-vision)

Happy to discuss it : especially, I am not knowledgeable in CI jobs so I do not know if its works.

### CODE REVIEW

**Review by MateoLostanlen — APPROVED** [review]
Thanks to bring it here too :)

File: docs/source/conf.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
-# Copyright (C) 2021, Pyronear contributors.
+# Copyright (C) 2019-2022, Pyronear.
```
Comment (frgfm) [inline docs/source/conf.py:1]: 2020-2022 ;)

**Review by frgfm — CHANGES_REQUESTED** [review]
Thanks a lot! There is only a small mistake on the copyright year :)

File: .github/validate_headers.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+from datetime import datetime
+from pathlib import Path
+
+shebang = ["#!usr/bin/python\n"]
+blank_line = "\n"
+
+# Possible years
+starting_year = 2019
```
[the remark this answers is not in the document]
Comment (fe51) [reply .github/validate_headers.py:8]: haha oops ! let's go for for commit on 30+ files :D

**Comment by fe51** [comment]
copyright year corrected :)

**Review by MateoLostanlen — APPROVED** [review]
thanks for the fix !

**Comment by fe51** [comment]
@frgfm I think we are good now !

**Review by frgfm — APPROVED** [review]
Cheers!

**Pull request merged**